In [2]:
import pandas as pd
import numpy as np
# import pydot

import matplotlib.pyplot as plt
import matplotlib
import seaborn as sns
from sklearn.model_selection import train_test_split

import statsmodels.formula.api as smf
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import GradientBoostingRegressor

from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error

from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.neural_network import MLPClassifier

from sklearn.svm import SVC

from sklearn.metrics import f1_score,roc_curve,auc,precision_score,recall_score

from sklearn.metrics import confusion_matrix

import seaborn as sns
from scipy import stats
import scipy.stats
import statsmodels.api as smd
from statsmodels.stats.proportion import proportions_ztest

# 가설검정, 분산분석하는 패키지
from statsmodels.formula.api import ols # 분산분석하기? 모델링?
from statsmodels.stats.anova import anova_lm # 아노바 데이터프레임 형태 출력
from scipy.stats import shapiro
%matplotlib inline
from sklearn.preprocessing import scale,minmax_scale,robust_scale
from scipy.stats import probplot
from sklearn.model_selection import train_test_split
from statsmodels.api import Logit

# font_path = 'C:/Windows/Fonts/gulim.ttc'
font_path = 'C:/Windows/Fonts/gulim.ttc'
font_name = matplotlib.font_manager.FontProperties(fname=font_path).get_name()
matplotlib.rc('font', family = font_name)

from matplotlib import font_manager,rc

from statsmodels.stats.outliers_influence import variance_inflation_factor
from sklearn.linear_model import LinearRegression
from sklearn.feature_selection import RFE
from sklearn.preprocessing import StandardScaler
from statsmodels.api import qqplot, add_constant

# from sklearn.tree import export_graphviz
# import graphviz

from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error
import numpy as np
# import pydot

import matplotlib.pyplot as plt
import matplotlib
import seaborn as sns
from sklearn.model_selection import train_test_split

import statsmodels.formula.api as smf
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import GradientBoostingRegressor

from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error

from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.neural_network import MLPClassifier

from sklearn.svm import SVC

from sklearn.metrics import f1_score,roc_curve,auc,precision_score,recall_score

from sklearn.metrics import confusion_matrix

import seaborn as sns
from scipy import stats
import scipy.stats
import statsmodels.api as smd
from statsmodels.stats.proportion import proportions_ztest

# 가설검정, 분산분석하는 패키지
from statsmodels.formula.api import ols # 분산분석하기? 모델링?
from statsmodels.stats.anova import anova_lm # 아노바 데이터프레임 형태 출력
from scipy.stats import shapiro
%matplotlib inline
from sklearn.preprocessing import scale,minmax_scale,robust_scale
from scipy.stats import probplot
from sklearn.model_selection import train_test_split
from statsmodels.api import Logit

# # font_path = 'C:/Windows/Fonts/gulim.ttc'
# font_path = '/usr/share/fonts/opentype/noto/NotoSerifCJK-ExtraLight.ttc'
# font_name = matplotlib.font_manager.FontProperties(fname=font_path).get_name()
# matplotlib.rc('font', family = font_name)

from matplotlib import font_manager,rc

from statsmodels.stats.outliers_influence import variance_inflation_factor
from sklearn.linear_model import LinearRegression
from sklearn.feature_selection import RFE
from sklearn.preprocessing import StandardScaler
from statsmodels.api import qqplot, add_constant

# from sklearn.tree import export_graphviz
# import graphviz

from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error

In [3]:
pay_1 = pd.read_excel('normal_final_classification.xlsx',encoding='euckr')
pay_2 = pd.read_excel('blood_final_classification.xlsx',encoding='euckr')

In [4]:
req = pd.read_csv('req_predict.csv',encoding='euckr')

In [5]:
req = req[['customer_id','risk']]

In [6]:
pay = pd.concat([pay_1,pay_2])[['customer_id','final_risk']]

In [7]:
d1 = pd.merge(req,pay,on='customer_id',how='inner')

In [8]:
d1['last_judge'] = d1['risk'] + d1['final_risk']

In [9]:
d2 = pd.merge(req,pay,on='customer_id',how='left')
d2 = d2[d2['final_risk'].isna() == True]

In [10]:
d2['final_risk'] = 0
d2

customer_id  risk  final_risk
0        C112358   1.0           0
1        C112374   1.0           0
5        C112426   1.0           0
6        C112427   1.0           0
13       C112510   1.0           0
...          ...   ...         ...
2728     C132806   1.0           0
2732     C132827   1.0           0
2800     C133991   1.0           0
2822     C134201   1.0           0
2824     C134204   1.0           0

[825 rows x 3 columns]

In [11]:
d2['final_risk'] = np.where(d2['risk']==1,d2['risk']+1,d2['final_risk'])
d2['final_risk'] = np.where(d2['risk']==2,d2['risk']+2,d2['final_risk'])
d2.groupby('final_risk').count()

customer_id  risk
final_risk                   
0.0                  33    33
2.0                 765   765
4.0                  27    27

In [12]:
d2['last_judge'] = d2['final_risk']
d2

customer_id  risk  final_risk  last_judge
0        C112358   1.0         2.0         2.0
1        C112374   1.0         2.0         2.0
5        C112426   1.0         2.0         2.0
6        C112427   1.0         2.0         2.0
13       C112510   1.0         2.0         2.0
...          ...   ...         ...         ...
2728     C132806   1.0         2.0         2.0
2732     C132827   1.0         2.0         2.0
2800     C133991   1.0         2.0         2.0
2822     C134201   1.0         2.0         2.0
2824     C134204   1.0         2.0         2.0

[825 rows x 4 columns]

In [13]:
d1

customer_id  risk  final_risk  last_judge
0        C112409   1.0           0         1.0
1        C112414   1.0           0         1.0
2        C112421   1.0           0         1.0
3        C112431   1.0           0         1.0
4        C112437   1.0           0         1.0
...          ...   ...         ...         ...
1997     C134163   1.0           0         1.0
1998     C134198   0.0           0         0.0
1999     C134203   1.0           0         1.0
2000     C134237   1.0           0         1.0
2001     C134240   1.0           0         1.0

[2002 rows x 4 columns]

In [14]:
judge_reject = pd.concat([d1,d2]).drop(['risk','final_risk'],axis=1)

In [15]:
judge_reject

customer_id  last_judge
0        C112409         1.0
1        C112414         1.0
2        C112421         1.0
3        C112431         1.0
4        C112437         1.0
...          ...         ...
2728     C132806         2.0
2732     C132827         2.0
2800     C133991         2.0
2822     C134201         2.0
2824     C134204         2.0

[2827 rows x 2 columns]

In [16]:
judge_reject.groupby('last_judge').count()

customer_id
last_judge             
0.0                  81
1.0                1822
2.0                 879
3.0                  17
4.0                  28

# 최종 고객 분류

- 최종 Score 0 : 81명, 수익, 유인 결정
- 최종 Score 1 : 1822명, 할증, 유인 결정(약한 수준의 할증)
- 최종 Score 2 : 879명, 엄격한 분류 기준을 도입(Screen 참고) 후 유인 또는 할증 결정 => 추가 개선 필요 사항
- 최종 Score 3 (17명), Score 4(28명) : 총 45명 거절 유지, 고위험군에 해당

In [17]:
judge_reject

customer_id  last_judge
0        C112409         1.0
1        C112414         1.0
2        C112421         1.0
3        C112431         1.0
4        C112437         1.0
...          ...         ...
2728     C132806         2.0
2732     C132827         2.0
2800     C133991         2.0
2822     C134201         2.0
2824     C134204         2.0

[2827 rows x 2 columns]

In [ ]:
df_raw = pd.read_csv('insu_pre_review')